### DISCUSSION TITLE
About gradient loss in the parameter space

### DISCUSSION THREAD

**Discussion by monlabiss** [mitsuba-renderer/mitsuba3#1020] [Q&A] [answered]
Hi :)
The below image is gradient loss image with respect to scene parameter(painting texture) in Veach door scene using the dr.forward() function.
I would like to understand how the gradient loss in the parameter space is actually calculated to obtain a gradient image.

Since the actual resolution of the painting texture is larger than the resolution of the painting in the image, a many-to-one relationship is created. Therefore, I would like to know how the calculation is done when obtaining such a gradient image or gradient values. Because it is hard to calculate because of correlation.

Please let me know both the theoretical explanation and how the code implementation is carried out or which parts of the code to look at.

![image](https://github.com/mitsuba-renderer/mitsuba3/assets/142881156/4b70a59a-8064-4547-88f9-d43245ed262c)

**Comment by rtabbara** [comment, marked as answer]
Hi @monlabiss,

A texture lookup that for instance uses bilinear interpolation is expressed as a weighted sum of neighbouring texels and hence the derivative of that lookup with respect to some parameter is similarly a weighted sum of derivatives of each texel with respect to the parameter. For a given pixel in the scene image, the incoming radiance could have contributions from multiple texture lookups due to light paths that hit the painting (potentially multiple times due to multiple bounces). Hence, regardless of the size of the painting in the scene, you can still have a "many-to-one" relationship.

If the question is more, from an optimisation point of view, is it difficult to recover for instance a high-resolution albedo texture when it's larger than the rendered view, the answer is yes however there are techniques such as regularization or [Large Steps](http://rgl.s3.eu-central-1.amazonaws.com/media/papers/Nicolet2021Large.pdf) that can help with convergence.

I'm sorry, I don't think I can provide a single piece of code that encompasses all of this but I think the suite of [Inverse rendering tutorials](https://mitsuba.readthedocs.io/en/latest/src/inverse_rendering_tutorials.html) can help to shed further light.

**Reply by monlabiss** [reply]
I understand! Many thanks.
